### PR TITLE
Validate phone numbers by country code with Laravel-Phone.

### LINKED ISSUE #472: BR number format suggestion invalid
The suggested number format in the vote submission form for Brazil does not validate.

image

but a submisison following the suggested phone number format produces a form submission error.

image

**Suggested Format**:
- https://en.wikipedia.org/wiki/Telephone_numbers_in_Brazil
- Typical format:
  - AA NNNN-NNNN
  - AA 9NNNN-NNNN

### PR DESCRIPTION
**Pull request by DFurnes**
#### Changes

Fixes #472 by using [Laravel-Phone](https://github.com/Propaganistas/Laravel-Phone) to validate phone numbers by the current session's country code. Laravel-Phone is a wrapper of Google's libphonenumber.
#### Current Issues

~~This seems to work great for US users and 10-digit BR numbers, but fails when given an 11-digit BR number. I'm still trying to figure out why, but that's probably a blocker for merging this in.~~
Should be fixed after some revisions made below!
#### How should this be tested?

Automated tests should continue to pass. It'd also be worthwhile to pull down this branch and test with various phone number & country code combinations to see if you can find any unexpected behavior (in addition to the issue mentioned above).

One thing I'm concerned about is that libphonenumber tries to be very _smart_. For example, it marked my `(123) 456-5555` test number as invalid. This is obviously really nice if it works, but I'm afraid there might be edge cases that'd be hard to track down... does anyone have feelings on that?

/cc @angaither @DeeZone

### CODE REVIEW

**Comment by deezone** [comment]
In terms of the Brazilian mobile numbers: https://en.wikipedia.org/wiki/Telephone_numbers_in_Brazil

> 10 digits (landlines and mobile)
> 11 digits (mobiles - area codes from 11-38 and 71-99)

We'll want to accept the 10 digit numbers from Brazil as well as accept the 11 digit numbers provided the area codes are in range.

> Typical format:
> - AA NNNN-NNNN
> - AA 9NNNN-NNNN

**Question**:
- Does the validation / submissions to Message Broker include the country code (MX: 52 / BR: 55) by `user_country`? Message Broker can take care of ensuring the country codes are in place for the submission to mGage but it would be a nice to have.

**Comment by angaither** [comment]
@DFurnes i know the 555 numbers was an issue we ran into DoSomething/phoenix/pull/2999

**Comment by angaither** [comment]
as far as other edge cases go, I almost always feel better when we use external validation, because I assume the other person is smarter at validating than I am

**Comment by DFurnes** [comment]
> We'll want to reject the 10 digit numbers from Brazil and accept the 11 digit numbers provided the area codes are in range.

@DeeZone Ah, that's interesting. There is a `mobile` flag I can include to have it try to "require" a number be associated with a cell phone. Let me see if that helps out.

> Does the validation / submissions to Message Broker include the country code (MX: 52 / BR: 55) by user_country? Message Broker can take care of ensuring the country codes are in place for the submission to mGage but it would be a nice to have.

@DeeZone At the moment, it only includes a sanitized (just numbers) version of the inputted number. I believe we _can_ use libphonenumber to add any implied country prefixes but that'd be an additional PR.

> i know the 555 numbers was an issue we ran into DoSomething/phoenix#2999

@angaither Yeah, and it _seems_ like libphonenumber is smart about this. I can definitely get some numbers with `555` in them to validate, but it's using some sort of ruleset that isn't entirely clear at first glance.

> as far as other edge cases go, I almost always feel better when we use external validation, because I assume the other person is smarter at validating than I am

@angaither Indeed!

**Comment by DFurnes** [comment]
Hmm, so for what it's worth:
- All Brazilian phone numbers we received in AGG 2015 validate correctly via the new validation code.
- ~~Only 2 of our Mexican numbers (from CGG 2014) validate correctly via the new validation code.~~ See the update posted below!

Not sure whether that means the data is bad or the validation is bad, but probably worth further thought.

**Comment by DFurnes** [comment]
And follow-up to that last post, when removing the `mobile` restriction on the validator, almost all of those "failing" Mexican numbers now validate. Phew! :smile:

**Comment by DFurnes** [comment]
@angaither Ready for review! :eyeglasses:

File: app/Http/Requests/VoteRequest.php [hunk 1]
```diff
@@ -63,8 +63,7 @@ public function rules()
     public function messages()
     {
         return [
-            'phone.phone' => 'That doesn\'t look like a real phone number!',
-            'birthdate.date' => 'Enter your birthday MM/DD/YYYY!',
```
Comment (angaither) [inline app/Http/Requests/VoteRequest.php:67]: just checking you want to remove this!

**Comment by angaither** [comment]
one :question: but otherwise, :+1:

File: app/Http/Requests/VoteRequest.php (hunk 1 above)
Comment (DFurnes) [inline app/Http/Requests/VoteRequest.php:67]: Yes, should've probably broken that into it's own commit. Date validation was changed in #474 to use the custom `localized_date` validator which returns the proper localized syntax, so this was just some leftover cruft.
